core/controllers/skill_editor.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the skill editor."""
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import skill_domain
+from core.domain import skill_services
+import feconf
+import utils
+
+
+def _require_valid_skill_id(skill_id):
+    """Checks whether the skill id received from the frontend is a
+    valid one.
+    """
+    if not isinstance(skill_id, basestring):
+        raise base.BaseHandler.InvalidInputException(
+            Exception('Skill id should be a string.'))
+
+    if len(skill_id) != 12:
+        raise base.BaseHandler.InvalidInputException(
+            Exception('The skill id given is invalid.'))
+
+
+class SkillEditorPage(base.BaseHandler):
+    """The editor page for a single skill."""
+
+    @acl_decorators.can_access_admin_page
+    def get(self, skill_id):
+        """Handles GET requests."""
+
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException()
```

Comment:
Is it `raise self.PageNotFoundException()` or  `raise self.PageNotFoundException`? I see both cases in your code.

Probably want to look at the rest of the codebase to see what we've done so far, and standardize. I think it's meant to be the latter actually (but please check).

Reply:
Done, changed it to  raise self.PageNotFoundException.